File: src/nodes/const_decl.zig
const nd = @import("node.zig");
const bytecode = @import("../common/bytecodes.zig");

pub const NodeConstDecl = struct {
    name: []const u8,
    vtype: bytecode.ApicaTypeBytecode,
    expression: *nd.Node,

    pub fn init(name: []const u8, vtype: bytecode.ApicaTypeBytecode, expr: *nd.Node) NodeConstDecl {
        return NodeConstDecl{ .name = name, .vtype = vtype, .expression = expr };
    }

    pub fn get_name(self: *const NodeConstDecl) []const u8 {
        return self.name;
    }

    pub fn get_vtype(self: *const NodeConstDecl) bytecode.ApicaTypeBytecode {
        return self.vtype;
    }

    pub fn get_expression(self: *const NodeConstDecl) *nd.Node {
        return self.expression;
    }
};

File: src/nodes/const_decl.rs
use apica_common::bytecodes::ApicaTypeBytecode;
use crate::nodes::node::Node;

pub struct NodeConstDecl {
    name: String,
    value_kind: ApicaTypeBytecode,
    expression: Node,
}

impl NodeConstDecl {
    pub fn init(name: String, value_kind: ApicaTypeBytecode, expression: Node) -> NodeConstDecl {
        NodeConstDecl { name, value_kind, expression }
    }
    
    pub fn get_name(&self) -> &String {
        &self.name
    }
    
    pub fn get_value_kind(&self) -> &ApicaTypeBytecode {
        &self.value_kind
    }
    
    pub fn get_expression(&self) -> &Node {
        &self.expression
    }
}